File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        vector<vector<int>> coord;

        for (int i=0; i<matrix.size(); i+=1){
            for (int j=0; j<matrix[0].size(); j+=1){
                if (matrix[i][j] == 0){
                    coord.push_back({i,j});
                }
            }
        }

        for (int i=0; i<coord.size(); i+=1){
            int r = coord[i][0];
            int c = coord[i][1];

            for (int j=0; j<matrix[0].size(); j+=1){
                matrix[r][j] = 0;
            }
            for (int j=0; j<matrix.size(); j+=1){
                matrix[j][c] = 0;
            }
        }
    }
};

File: 73-set-matrix-zeroes/set-matrix-zeroes.go
func setZeroes(matrix [][]int)  {
    cell := matrix[0][0]
    m, n := len(matrix), len(matrix[0])
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            if matrix[i][j] == 0 {
                if j == 0 {
                    cell = 0
                }
                matrix[i][0] = 0
            }
        }
    }

    for i := 0; i < m; i++ {
        if matrix[i][0] == 0 {
            // row
            for j := 1; j < n; j++ {
                if matrix[i][j] == 0 {
                    // col
                    for k := 0; k < m; k++ {
                        matrix[k][j] = 0
                    }
                } else {
                    matrix[i][j] = 0
                }
            }
        }
    }

    if cell == 0 {
        for i := 0; i < m; i++ {
            matrix[i][0] = 0
        }
    }
}